__tests__/integration/utils/test-utils-integration.js

```diff
@@ -855,38 +865,109 @@ export class TestUtils {
    *                          or rejects if timeout is reached
    */
   static async waitForTxReceived(walletId, txId, timeout) {
-    /* eslint-disable no-async-promise-executor */
-    return new Promise(async (resolve, reject) => {
-      let timeoutHandler;
-      let timeoutReached = false;
-      if (timeout) {
-        // Timeout handler
-        timeoutHandler = setTimeout(() => {
-          timeoutReached = true;
-        }, timeout);
-      }
+    let timeoutHandler;
+    let timeoutReached = false;
+    if (timeout) {
+      // Timeout handler
+      timeoutHandler = setTimeout(() => {
+        timeoutReached = true;
+      }, timeout);
+    }
 
-      while ((await TestUtils.getTransaction(walletId, txId)).success === false) {
-        if (timeoutReached) {
-          break;
-        }
-        // Tx not found, wait 1s before trying again
-        await delay(1000);
+    let result = await TestUtils.getTransaction(walletId, txId);
+
+    while (result.success === false || result.processingStatus !== 'finished') {
```

Comment:
I wanted to use the ENUM from the lib but we are not exporting the full types file, so I couldn't. Maybe we should export all types.